lib/recipe.rb

```diff
@@ -30,24 +31,26 @@
 
 # only support solaris 10 and 11
 return if platform?('solaris2') && node['platform_version'] !~ /^5.(10|11)$/
+
 liveness_agent = <<'AUTOMATE_LIVENESS_AGENT'
 #LIVENESS_AGENT
 AUTOMATE_LIVENESS_AGENT
 liveness_agent.gsub!('#!/usr/bin/env ruby', "#!#{Gem.ruby}")
 
-agent_dir         = Chef::Config.platform_specific_path(
```

Comment:
That comment was actually thinking you were referring to the log platform specific path, but the same basic idea applies to both. We're handing windows and non-windows explicitly so we don't need that helper.